labelbox/data/annotation_types/data/tiled_image.py

```diff
@@ -0,0 +1,460 @@
+                                dtype=np.int)
+        #convert geo bounds to pixel bounds. assumes geo bounds are in wgs84/EPS4326 per leaflet
+        global_bounds = np.array([
+            PygeoPoint.from_latitude_longitude(latitude=point.y,
+                                               longitude=point.x).pixels(zoom)
+            for point in geo_bounds
+        ])
+
+        #get the range of pixels for both sets of bounds to use as a multiplification factor
+        local_x_range, local_y_range = self._get_ranges(local_bounds)
+        global_x_range, global_y_range = self._get_ranges(global_bounds)
+
+        if src_epsg == EPSG.SIMPLEPIXEL:
+
+            def transform(x: int, y: int):
+                scaled_xy = (x * (global_x_range) / (local_x_range),
+                             y * (global_y_range) / (local_y_range))
+
+                minx, _, miny, _ = self._min_max_x_y(global_bounds)
+                x, y = map(lambda i, j: i + j, scaled_xy, (minx, miny))
+
+                point = PygeoPoint.from_pixel(pixel_x=x, pixel_y=y,
+                                              zoom=zoom).latitude_longitude
+                #convert to the desired epsg
+                return Transformer.from_crs(EPSG.EPSG4326.value,
+                                            geo_bounds_epsg.value,
+                                            always_xy=True).transform(
+                                                point[1], point[0])
+
+            self.transform_function = transform
+
+        #geo to pixel - converts a point in geo coords to pixel coords
+        #handles 4326 from lat,lng
+        elif src_epsg == EPSG.EPSG4326:
+
+            def transform(x: int, y: int):
+                point_in_px = PygeoPoint.from_latitude_longitude(
+                    latitude=y, longitude=x).pixels(zoom)
+
+                minx, _, miny, _ = self._min_max_x_y(global_bounds)
+                x, y = map(lambda i, j: i - j, point_in_px, (minx, miny))
+
+                return (x * (local_x_range) / (global_x_range),
+                        y * (local_y_range) / (global_y_range))
+
+            self.transform_function = transform
+
+        #handles 3857 from meters
+        elif src_epsg == EPSG.EPSG3857:
+
+            def transform(x: int, y: int):
+                point_in_px = PygeoPoint.from_meters(meter_y=y,
+                                                     meter_x=x).pixels(zoom)
+
+                minx, _, miny, _ = self._min_max_x_y(global_bounds)
+                x, y = map(lambda i, j: i - j, point_in_px, (minx, miny))
+
+                return (x * (local_x_range) / (global_x_range),
+                        y * (local_y_range) / (global_y_range))
+
+            self.transform_function = transform
+
+    def __call__(self, point: Point):
+        if self.transform_function is not None:
+            res = self.transform_function(point.x, point.y)
+            return Point(x=res[0], y=res[1])
```

Comment:
This should accept any vector geometry (non-mask) and then project each point. and return the updated geometry right? Or was that going to happen in the geometry?